drivers/elasticsearch/src/main/java/io/sbk/driver/Elasticsearch/ElasticsearchReader.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) KMG. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.sbk.driver.Elasticsearch;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import co.elastic.clients.elasticsearch._types.ElasticsearchException;
+import co.elastic.clients.elasticsearch.core.GetRequest;
+import co.elastic.clients.elasticsearch.core.GetResponse;
+import io.sbk.params.ParameterOptions;
+import io.sbk.api.Reader;
+import io.sbk.system.Printer;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Class for Elasticsearch Reader.
+ */
+public class ElasticsearchReader implements Reader<String> {
+    private final ElasticsearchConfig config;
+    private final ElasticsearchClient client;
+    private long id;
+
+    public ElasticsearchReader(int readerId, ParameterOptions params, ElasticsearchConfig config, ElasticsearchClient client) {
+        this.id = Elasticsearch.generateStartKey(readerId);
+        this.config = config;
+        this.client = client;
+    }
+
+    @Override
+    public String read() throws IOException {
+        try {
+            GetRequest request = GetRequest.of(g -> g
+                    .index(config.index.trim())
+                    .id(String.valueOf(id++))
+            );
+            GetResponse<Map> response = client.get(request, Map.class);
+            return response.fields().toString();
+        } catch (ElasticsearchException e) {
+            Printer.log.error("Elastic Search: recordRead failed !");
+            throw new IOException(e);
+        }
+    }
+
+    @Override
+    public void close() throws IOException {
+        shutDownElasticsearch();
```

Comment:
dont call while reader is exiting? shutdown entire elastic search jus for a sake one writer is not good.  remove this line ,